contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -347,30 +347,63 @@ library VaultLifecycleSTETH {
         address crvPool,
         uint256 minETHOut
     ) external returns (uint256) {
+        require(
+            amount >= minETHOut,
+            "Amount withdrawn smaller than minETHOut from swap"
+        );
+
         uint256 assetBalance = address(this).balance;
 
         uint256 amountETHOut = DSMath.min(assetBalance, amount);
 
+        // We pass in the amount of stETH we want to unwrap from wstETH
+        // Though stETH != ETH, we assume that they are equivalent here
+        // by passing in the amount of ETH we need to withdraw
+        // This assumption is fine because we will be swapping the stETH to ETH.
+        uint256 stethNeeded =
+            DSMath.max(assetBalance, amount).sub(assetBalance);
         uint256 amountToUnwrap =
-            IWSTETH(collateralToken).getWstETHByStETH(
-                DSMath.max(assetBalance, amount).sub(assetBalance)
-            );
+            IWSTETH(collateralToken).getWstETHByStETH(stethNeeded);
 
         if (amountToUnwrap > 0) {
             IWSTETH wsteth = IWSTETH(collateralToken);
-            // Unrap to stETH
+            IERC20 steth = IERC20(wsteth.stETH());
+
+            uint256 startStethBalance = steth.balanceOf(address(this));
+
+            // Unwrap to stETH
             wsteth.unwrap(amountToUnwrap);
 
+            // Post-unwrap, the stETH balance will not completely match the stethNeeded
+            // due to precision issues.
```

Comment:
if we consistently know what the inconsistency will be we can just keep the way it is and not add more gas for user. but this is more dependable.